### ISSUE TITLE
Doesn't compile with c++11

### ISSUE THREAD

**Issue by davispuh** [scim-im/scim#6]
GCC 6.0 defaults to `c++11`, `-std=c++11` can be passed to 5.x and it fails to compile with error
```
In file included from /usr/include/c++/6.0/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/6.0/bits/char_traits.h:39,
                 from /usr/include/c++/6.0/ios:40,
                 from /usr/include/c++/6.0/ostream:38,
                 from /usr/include/c++/6.0/iostream:39,
                 from .././../../../src/scim.h:193,
                 from .././../../../src/scim_backend.cpp:33:
/usr/include/c++/6.0/bits/stl_pair.h: In instantiation of ‘struct std::pair<const std::__cxx11::basic_string<char>, scim::Pointer<scim::IMEngineFactoryBase> >’:
.././../../../src/scim_backend.cpp:117:82:   required from here
/usr/include/c++/6.0/bits/stl_pair.h:191:17: error: ‘constexpr std::pair<_T1, _T2>::pair(const std::pair<_T1, _T2>&) [with _T1 = const std::__cxx11::basic_string<char>; _T2 = scim::Pointer<scim::IMEngineFactoryBase>]’ declared to take const reference, but implicit declaration would take non-const
       constexpr pair(const pair&) = default;
                 ^

```

**Comment by tzhuan** [comment]
Thanks for the bug report. I don't have gcc6 in hand but I believe it is solved in bdeff0b45eda2e24e7f0221e54e77f89307c6258 . Could you please try this? Thank you.

**Comment by davispuh** [comment]
Cool and thanks for quick fix, it indeed fixed this error. Only got another but that was very simple fix, see #7 and then it compiles successfully.
Also you don't need GCC 6.0, but I think any 5 version with `-std=c++11` (or `-std=c++0x`) should do.

**Comment by tzhuan** [comment]
Well, I don't have gcc-5 in hand either. I tested it with gcc-4.9.2 with -std=c++11 and it passed. :-)
